client-java/src/main/java/uk/gov/gchq/palisade/client/java/JavaClient.java

```diff
@@ -0,0 +1,225 @@
+
+import uk.gov.gchq.palisade.client.java.download.DownloadFailedEvent;
+import uk.gov.gchq.palisade.client.java.download.DownloadManager;
+import uk.gov.gchq.palisade.client.java.download.DownloadTracker;
+import uk.gov.gchq.palisade.client.java.job.JobConfig;
+import uk.gov.gchq.palisade.client.java.job.JobContext;
+import uk.gov.gchq.palisade.client.java.request.Context;
+import uk.gov.gchq.palisade.client.java.request.PalisadeClient;
+import uk.gov.gchq.palisade.client.java.request.PalisadeRequest;
+import uk.gov.gchq.palisade.client.java.request.RequestId;
+import uk.gov.gchq.palisade.client.java.request.UserId;
+import uk.gov.gchq.palisade.client.java.resource.ResourceClient;
+import uk.gov.gchq.palisade.client.java.resource.ResourceReadyEvent;
+import uk.gov.gchq.palisade.client.java.resource.ResourcesExhaustedEvent;
+import uk.gov.gchq.palisade.client.java.util.Bus;
+
+import javax.inject.Singleton;
+import javax.websocket.ContainerProvider;
+import javax.websocket.DeploymentException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+
+import static uk.gov.gchq.palisade.client.java.job.IJobContext.createJobContext;
+
+/**
+ * <p>
+ * The main client implementation.
+ * </p>
+ * <p>
+ * Note that this class should only be created via the static {@code create}
+ * methods on the {@link Client} interface. For tests it can be injected into a
+ * test class annotated with {@code @MicronautTest} as it is configured for
+ * dependency injection. This makes testing much easier.
+ *
+ * @since 0.5.0
+ */
+@Singleton
+public class JavaClient implements Client {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JavaClient.class);
+    private static final String EVENT_CAUGHT = "Caught event: {}";
+
+    private final ClientContext clientContext;
+    private final Map<String, JobContext> jobs = new HashMap<>();
+
+    /**
+     * Create a new JavaClient injected with required services. The main injection
+     * is the actual ApplicationContext that is performing the injection. Same as
+     * Guice, Micronaut can inject itself, which is cool.
+     *
+     * @param clientCcontext  The {@code ApplicationContext} doing the injecting
+     */
+    public JavaClient(final ClientContext clientCcontext) {
+        this.clientContext = Objects.requireNonNull(clientCcontext);
+    }
+
+    @Override
+    @SuppressWarnings("java:S3242") // I REALLY want to use UnaryOperator here SonarQube!!!
+    public Result submit(final UnaryOperator<JobConfig.Builder> func) {
+        return submit(func.apply(JobConfig.builder()).build());
+    }
+
+    @Override
+    public Result submit(final JobConfig jobConfig) {
+
+        LOG.debug("Job configuration submitted: {}", jobConfig);
+
+        final var palisadeRequest = createRequest(jobConfig); // actual instance sent to palisade Service
```

Comment:
Thought we agreed not to use var? Just think it makes the code a little more awkward.

Reply:
I'm sure it was brought up and we could use all Java 11 language features.